Budget Blocks/Budget Blocks/Networking/NetworkingController.swift

```diff
@@ -86,24 +86,33 @@ class NetworkingController {
                 NSLog("No data returned from login request")
                 return completion(nil, error)
             }
-            
-            do {
-                let responseJSON = try JSON(data: data)
-                if let token = responseJSON["token"].string,
-                    let userID = responseJSON["id"].int,
-                    let linked = responseJSON["LinkedAccount"].bool,
-                    let manual = responseJSON["ManualOnly"].bool {
-                    self.bearer = Bearer(token: token, userID: userID, linkedAccount: linked, manualAccount: manual)
-                    self.keychain.set(email, forKey: self.emailKey)
-                    self.keychain.set(password, forKey: self.passwordKey)
-                }
-                completion(responseJSON["token"].string, nil)
-            } catch {
-                completion(nil, error)
-            }
+            completion(self.jsonData(data: data, email: email, password: password), nil)
         }.resume()
     }
     
+    func jsonData(data: Data, email: String,password: String) -> String {
+        let _: JSON = ["email": email, "password": password]
+        let myToken = ""
+        
+        do {
+            let responseJSON = try JSON(data: data)
+            if  let token = responseJSON["token"].string,
+                let userID = responseJSON["id"].int,
+                let linked = responseJSON["LinkedAccount"].bool,
+                let manual = responseJSON["ManualOnly"].bool {
+                self.bearer = Bearer(token: token, userID: userID, linkedAccount: linked, manualAccount: manual)
+                self.keychain.set(email, forKey: self.emailKey)
+                self.keychain.set(password, forKey: self.passwordKey)
+            }
+            guard let myToken = responseJSON["token"].string else { fatalError() }
+            return myToken
+        } catch {
+            print("logging error: \(error.localizedDescription)")
```

Comment:
Can we make sure to add a user friendly error handler here, especially once we move to production? We talked about possibly giving an alert to the user to let them know there was a login error.